core/arch/arm/pta/system.c

```diff
@@ -40,6 +41,66 @@ static TEE_Result system_rng_reseed(struct tee_ta_session *s __unused,
 	return TEE_SUCCESS;
 }
 
+static TEE_Result system_derive_ta_unique_key(struct tee_ta_session *s,
+					      uint32_t param_types,
+					      TEE_Param params[TEE_NUM_PARAMS])
+{
+	size_t data_len = sizeof(TEE_UUID);
+	TEE_Result res = TEE_ERROR_GENERIC;
+	uint8_t *data = NULL;
+	uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+					  TEE_PARAM_TYPE_MEMREF_OUTPUT,
+					  TEE_PARAM_TYPE_NONE,
+					  TEE_PARAM_TYPE_NONE);
+
+	if (exp_pt != param_types) {
+		res = TEE_ERROR_BAD_PARAMETERS;
```

Comment:
On second thought, if `params[1]` isn't a memref, then we can't clear it.

Reply:
True, i'll keep `TEE_ERROR_BAD_PARAMETERS`, regarding the other parameter checks below. If we cannot guarantee the size of the buffer, then we cannot really know how much we could wipe. So, it seems like we're back to my original patch with regards to this, agree?

Reply:
Sure, this wiping is just a precaution to make it easier to spot errors.

Reply:
Reverted back (https://github.com/OP-TEE/optee_os/pull/3053/commits/07036eefb6c14335785c4991de1df53a0d6824b1).